lib/services/local/users.go

```diff
@@ -1354,6 +1359,12 @@ func (s *IdentityService) DeleteMFADevice(ctx context.Context, user, id string)
 	}
 
 	err := s.Delete(ctx, backend.NewKey(webPrefix, usersPrefix, user, mfaDevicePrefix, id))
+	if trace.IsNotFound(err) {
+		if _, err := s.getSSOMFADevice(ctx, user); err == nil {
+			return trace.BadParameter("cannot delete ephemeral SSO MFA device")
```

Comment:
Could we remove the guards from the service layer and rely on this instead? Does that make sense, or does it make the service layer worse?

Reply:
My thought is that I'd rather fail early in the service layer, no reason to go to the storage layer if we know it must fail. I only added it here for redundancy / for the potential of other callers to `DeleteMFADevice`. WDYT?

Reply:
In general I prefer for validation to have an "owner", so when it changes we don't need to chase it all over (or have silly failures because we over-validate all over). Here I would say storage is the owner, as it's the more fundamental layer.

That said, for this particular case, I think it's fine to keep service. It likely avoids other silly errors like saying "you can't delete <SSO device> because it's the last", which is just plain wrong.